### ISSUE TITLE
Bug: mypy incorrect error: "Missing return statement"

### ISSUE THREAD

**Issue by vfrazao-ns1** [python/mypy#5360]
Hello,

Upon running mypy on one of my files it claims that I am missing a return statement (incorrectly), sample code (starting line is #126):

```Python
    def _get_session_id(self, file_path: str) -> bytes:
        """
        The session ID is unique every day. Simply denotes the day. We use this
        to build the IEX Transport Protocol header.
        """
        try:
            return self.session_id
        except AttributeError:
            iex_header_start = (
                self.version
                + self.reserved
                + self.protocol_id
                + self.channel_id
            )
            with open(file_path, "rb") as market_file:
                for line in market_file:
                    if iex_header_start in line:
                        line = line.split(iex_header_start)[1]
                        return line[:4]
```

mypy error message: `126: error: Missing return statement`

I am using mypy 0.620 and Python 3.7. I would expect mypy to recognize that barring an unforeseen/uncaught exception the method will always have a return value.

**Comment by ethanhs** [comment]
Mypy is correct here.

If the `session_id` attribute does not exist (going to the `except` clause), and the `market_file` does not contain `iex_header_start`, then the for loop will be exhausted and there will be a default `None` return.
